lib/src/value/calculation.dart

```diff
@@ -158,6 +344,201 @@ class SassCalculation extends Value {
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    if (dividend is! SassNumber ||
+        modulus is! SassNumber ||
+        _arePossiblyCompatible(dividend, modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    _verifyCompatibleNumbers(args);
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    var strategyOrNumberSimplified = _simplify(strategyOrNumber);
+    numberOrStep = numberOrStep.andThen(_simplify);
+    step = step.andThen(_simplify);
+
+    switch ((strategyOrNumberSimplified, numberOrStep, step)) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassString(isVar: false) || SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(SassString('nearest'), number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when {'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy, number, step);
+
+      case (SassString(), _, _):
+      case (CalculationOperation(), _, _):
+      case (SassNumber(), _, null):
+        return SassCalculation._("round", [
+          strategyOrNumberSimplified,
+          if (numberOrStep != null) numberOrStep,
+          if (step != null) step
+        ]);
```

Comment:
Good point. Although, I'm not sure how I would be able to assign a variable for pattern matching properties like `isVar`. So for example: 
 ```
     case (SassString strategy, SassString(isVar: true), _):
        return SassCalculation._("round", [
          strategy,
          if (numberOrStep != null) numberOrStep,
          if (step != null) step
        ]);
```
In this case I would not be able to use a variable for numberOrStep. Also, If I'd wanted to declare a variable for the first argument I would have to separate all of the following cases and the `return SassCalculation...` block would become repetitive:
```
      case (SassString strategy, SassString(isVar: true), _):
      case (CalculationOperation() , _, null):
      case (SassNumber() , _, null):
```
This is because of the error `dynamic variable` I mention [here](https://github.com/sass/dart-sass/pull/1970#discussion_r1256213928).

Reply:
First of all, I don't think you should be handling the three-argument (or 2-with-var) case and the less-than-three argument cases in the same condition. Even if you can squeeze the code into shape, those conditions have different meanings for the arguments, so it's confusing to the reader.

Second, a trick I learned recently is that you can use `&&` to run a pattern match _and_ separately assign to a variable. So let's start by splitting out the three-argument case:

```dart
case (SassString strategy, SassString(isVar: true) && var number, var step):
  return SassCalculation._("round", [strategy, number, if (step != null) step]);
```

We can use a similar trick for the two-argument case:

```dart
case (
    (CalculationOperation() || SassNumber()) && var number?,
    var step?,
    null
  ):
  return SassCalculation._("round", [number, step]);
```

Reply:
Got it, I refactored the switch statement based on this comment, please let me know if I missed anything!